Live/Classes/ResourceLayer.cpp

```diff
@@ -0,0 +1,52 @@
+#include "ResourceLayer.h"
+#include <ctime>
+#include <cstdlib>
+
+USING_NS_CC;
+
+ResourceLayer::ResourceLayer() : Layer() {
+    srand ( time(NULL) );
+
+    // Note that resources can be generated on top of each other no issue, 
+    //      just watch out for collisions with map file's collision layer
+    for(int i = 0; i < MAX_RESOURCES; i++) {
+        // TODO: Implement collision detection with main map during generation
+        //  Randomly generate resource locations based on demo map
+        //  Currently hardcoded to ensure no collision with water or beach areas
+        int r = rand() % 12500;
+        float x = r/10;
+        r = rand() % 8600;
+        float y = r/10;
+
+        while(x >= 1050 && y <= 380) {
+            r = rand() % 12500;
+            x = r/10;
+            r = rand() % 8600;
+            y = r/10;
+        }
+
+        m_resources[i] = new ResourceObstacle(x,y);
+        this->addChild(m_resources[i]->getSprite());
+    }
+}
+
+bool ResourceLayer::checkCollision(cocos2d::Point position) {
+    // TODO: Replace point with rect, update MainScene.cpp to pass in a rect
+    Rect playerHitbox = Rect(position.x-2,position.y-24, 4, 12);
+    for(int i = 0; i < MAX_RESOURCES; i++) {
+        if(m_resources[i]->checkCollision(playerHitbox)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool ResourceLayer::gather(cocos2d::Point position, Direction dir) {
+    for(int i = 0; i < MAX_RESOURCES; i++) {
+        if(m_resources[i]->gather(position, dir)) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
lots of magic numbers lol, get rid of those.

Reply:
i've added variables for a couple that will matter, but the rest are hacks, i just needed them to avoid the body of water on the bottom right of the demo map